mp/src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -107,49 +109,67 @@ void CNEOHud_FriendlyMarker::DrawNeoHudElement()
 void CNEOHud_FriendlyMarker::DrawPlayerForTeam(C_Team* team, const C_NEO_Player* localPlayer) const
 {
 	auto memberCount = team->GetNumPlayers();
+	auto teamColour = GetTeamColour(team->GetTeamNumber());
 	for (int i = 0; i < memberCount; ++i)
 	{
-		auto player = team->GetPlayer(i);
-		auto teamColour = GetTeamColour(team->GetTeamNumber());
+		auto player = dynamic_cast<C_NEO_Player *>(team->GetPlayer(i));
```

Comment:
I think it's a safe assumption that if we ever get a valid `C_BasePlayer*`, it will be castable into a `C_NEO_Player*` (because that should be the only kind of player we have), so this could be `static_cast` for perf.